sunpy/map/maputils.py

```diff
@@ -375,3 +375,28 @@ def on_disk_bounding_coordinates(smap):
     return SkyCoord([np.nanmin(tx), np.nanmax(tx)] * u.arcsec,
                     [np.nanmin(ty), np.nanmax(ty)] * u.arcsec,
                     frame=smap.coordinate_frame)
+
+
+def contains_coordinate(smap, coordinates):
+    """
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A map in helioprojective Cartesian coordinates.
+
+    coordinates : `~astropy.coordinates.SkyCoord`, `~sunpy.coordinates.frames.Helioprojective`
+        The input coordinate. The coordinate frame must be
+        `~sunpy.coordinates.Helioprojective`.
+
+    Returns
+    -------
+    `~bool`
+        Returns `True` If coordinates are within the coordinate
+        bound of the smap and Returns `False` If coordinates are
+        not within the coordinate bound of the smap.
+    """
+    #Dimensions of smap
+    xs, ys = smap.dimensions
+    # Converting coordinates to pixels
+    xc, yc = smap.world_to_pixel(coordinates)
+    return xc<=xs and yc<=ys
```

Comment:
```suggestion
    return xc >= -0.5 and xc <= xs-0.5 and and yc >= -0.5 and yc <= ys-0.5
```
You need to check that the pixel is within the bounds of the other two edges as well, and you need to account for half pixels.  For example, if there are 1024 pixels along a particular dimension, the pixel numbers can go from 0 to 1023, but a coordinate resolves to within the map if its converted pixel value is within -0.5 to 1023.5.